x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/predicate/operator/comparison/In.java

```diff
@@ -105,6 +108,26 @@ protected Pipe makePipe() {
         return new InPipe(source(), this, children().stream().map(Expressions::pipe).collect(Collectors.toList()));
     }
 
+    @Override
+    protected TypeResolution resolveType() {
+        if (value instanceof FieldAttribute) {
+            try {
+                ((FieldAttribute) value).exactAttribute();
+            } catch (MappingException ex) {
+                return new TypeResolution(format(null, "[{}] cannot operate on first argument field of data type [{}]",
+                    functionName(), value().dataType().esType));
+            }
+        }
+
+        Optional<Expression> firstNotFoldable = list.stream().filter(expression -> !expression.foldable()).findFirst();
```

Comment:
I think this is more readable through a basic for loop and early return
```java
for (Expression ex : list) {
    if(ex.foldable() == false) {
         return new TypeResolution...
    }
}
```

About the same amount of lines, straight-forward and faster.